src/functions/helpers.ts

```diff
@@ -298,6 +298,78 @@ export function reduceNumbersTextAs0(
   );
 }
 
+// -----------------------------------------------------------------------------
+// FLATTEN
+// -----------------------------------------------------------------------------
+
+/**
+ * Flatten an array of items, where each item can be a single value or a 2D array, and apply the
+ * callback to each element.
+ *
+ * The 2D array are flattened row first.
+ */
+export function flattenRowFirst<T>(items: Array<T | Matrix<T>>): T[] {
+  /**/
+  return reduceAny(
+    items,
+    (array: T[], val: T) => {
+      array.push(val);
+      return array;
+    },
+    [],
+    "rowFirst"
+  );
+  /*/
+  //TODO : Check which "method" is faster or more readable
```

Comment:
@LucasLefevre @hokolomopo : Do you have an insight/idea of which method we should keep ?

Reply:
If we keep recudeAcy, we should definitevly put a bit of docstring on it, bacause at first glance it's hard to know that it handles matrices and simple arguments